mongo/src/org/immutables/mongo/repository/RepositorySetup.java

```diff
@@ -128,7 +127,7 @@ public String translateName(Member member) {
     class GsonNamingStrategy implements FieldNamingStrategy {
```

Comment:
I would move this class one level up it is already inside `RepositorySetup.FieldNamingStrategy`.

On the other hand, `FieldNamingStrategy` is already Single Abstract Method (SMA) interface so having a lambda on the top of `Gson` instance should be trivial for clients of this API.

Up to you ...

Reply:
I've been thinking about it for a short while. Seems that we better have something like `GsonNamingStrategy` public (or private with public factory method `namingStrategyFrom(Gson)`) because it sort of documents that it's possible and should be used sometimes. It is easy implement with lambda, but people might have no clue that they ought to do that and do smth hardcoded.

Reply:
Perhaps `GsonNamingStrategy` warrants its own file and be public or, at least, `gsonNamingStrategy` factory method (also public).

Not 100% sure if it should be inside `bson4gson` package / `GsonCodes` class.

Reply:
Maybe, but given that we think criteria is a newer implementation and to minimize impact of changes (as it is the goal of some changes asked for this PR) let it be just public `GsonNamingStrategy`, just can be moved out from interface and be directly nested in `RepositorySetup`  as `RepositorySetup.GsonNamingStrategy`. I think this will be minimal and cute enough.
P.S. I'll answer on other comments tomorrow morning.

Reply:
made it `public` visibility and moved to `RepositorySetup`